R/exclude.R

```diff
@@ -228,56 +231,34 @@ normalize_exclusions <- function(x, normalize_path = TRUE,
            " are not!",
            call. = FALSE)
     }
-    # x is a character vector (or list) of file names
-    # Normalize to list(<filename> = list(Inf), ...)
-    x <- structure(rep(list(Inf), length(x)), names = x)
-  } else {
-    unnamed <- names(x) == ""
-    if (any(unnamed)) {
-
-      # must be character vectors of length 1
-      bad <- vapply(
-        seq_along(x),
-        function(i) {
-          unnamed[i] && (!is.character(x[[i]]) || length(x[[i]]) != 1L)
-        },
-        logical(1L)
-      )
-
-      if (any(bad)) {
-        stop("Full file exclusions must be character vectors of length 1. items: ",
-             toString(which(bad)),
-             " are not!",
-             call. = FALSE)
-      }
-      names(x)[unnamed] <- x[unnamed]
-      x[unnamed] <- rep_len(list(list(Inf)), sum(unnamed))
-    }
-
-    full_line_exclusions <- !vapply(x, is.list, logical(1L))
+    # Normalize unnamed entries to list(<filename> = list(Inf), ...)
+    names(x)[unnamed] <- x[unnamed]
+    x[unnamed] <- rep_len(list(list(Inf)), sum(unnamed))
+  }
 
-    if (any(full_line_exclusions)) {
+  full_line_exclusions <- !vapply(x, is.list, logical(1L))
 
-      # must be integer or numeric vectors
-      bad <- vapply(
-        seq_along(x),
-        function(i) {
-          full_line_exclusions[i] && !is.numeric(x[[i]])
-        },
-        logical(1L)
-      )
+  if (any(full_line_exclusions)) {
 
-      if (any(bad)) {
-        stop("Full line exclusions must be numeric or integer vectors. items: ",
-             toString(which(bad)),
-             " are not!",
-             call. = FALSE)
-      }
+    # must be integer or numeric vectors
+    bad <- vapply(
```

Comment:
maybe better to just get the `vapply(x, is.numeric)` vector & compare with `full_line_exclusions` as a vector?